src/storage/src/hummock/file_cache/file.rs

```diff
@@ -82,6 +86,9 @@ impl CacheFile {
     pub async fn open(path: impl AsRef<Path>, options: CacheFileOptions) -> Result<Self> {
         options.assert();
 
+        assert!(options.fs_type != FsType::Tmpfs,
+            "Attempting to create cache file on a TmpFS file system. TmpFS cannot be used because it does not support Direct IO.");
```

Comment:
The helpful guidance in the developer guide is nice, wonder if you'd like to also put it into the assertion message?

Reply:
The bit about using an environment variable to override the location of the cache file? That guidance really only applies if running unit tests, so I could put something like `"Attempting to create cache file on a TmpFS file system. TmpFS cannot be used because it does not support Direct IO. If you are running unit tests, you can use the env var RISINGWAVE_TEST_DIR to specify a valid location for the cache file."`